### ISSUE TITLE
Please use the API for comicake

### ISSUE THREAD

**Issue by chocolatkey** [riderkick/FMD#1669]
Hi,
I am the creator of ComiCake, which champion scans, among others uses. I would like to ask, for everyone's sake, that you use the API instead of scraping the HTML on the page. The HTML is subject to change, especially since I am making multiple, completely different themes, while the API will remain very similar across releases and sites. Using champion scans as an example, the API is here: https://reader.championscans.com/api/. It is pretty easy to use, and you can see an example of how I implemented Tachiyomi support here: https://github.com/chocolatkey/tachiyomi-extensions/blob/master/src/all/comicake/src/eu/kanade/tachiyomi/extension/en/comicake/ComiCake.kt . If you have any questions regarding its usage, feel free to ask

**Comment by ChocolateOtaku** [comment]
> Using champion scans as an example...

**Fun Facts:**

1. The website itself does not use the API, pages are rendered on server side

2. Manga directory
JSON Requests: 2
HTML Requests: 3

3. Manga Directory
JSON Size: 35.1 KB (seems server not even bother to compress it during transfer)
HTML Size: 16.4 KB

4. Cannot copy API links directly from the website to paste them in FMD

5. FMD is not actively developed anymore, so changes will not happen, no matter what benefit the API has to offer

6. I wasted my time to respond to something that nobody cares about :(

----
<img width="796" alt="json" src="https://user-images.githubusercontent.com/38581381/52504221-0e125400-2be8-11e9-9a5e-e85d118c39ab.png">
<img width="768" alt="html" src="https://user-images.githubusercontent.com/38581381/52504219-0e125400-2be8-11e9-97c7-cc0d14c666a3.png">

**Comment by chocolatkey** [comment]
@ChocolateOtaku yeah the website doesn't use the API for everything because it doesn't need to. I like to make things work without globs of javascript everywhere. With the theme championscans has, only the reader and search uses the API, and that's all that needs to use the API. As for the compression, that's championscans' issue to resolve. They are not behind cloudflare and have also not configured their nginx setup properly. Nothing to do with comicake. Even then, FMD is a desktop application, not mobile. I think 15KB can be sacrificed for the sake of maintainability. I'm trying to do you a favor, because the templates are *not* going to be the same for every site. If you like scraping HTML and frequently updating due to theme changes that much than so be it, but you're on your own.

> FMD is not actively developed anymore, so changes will not happen, no matter what benefit the API has to offer

If FMD is not being actively developed anymore, then how was comicake added, and why is the last commit 29 days ago?

**Comment by ChocolateOtaku** [comment]
You didn't provide a solution for 4.)

29 days is very very inactive for a software that requires weekly updates to maintain all websites and keep them working (34 new issues within the last 29 days).

I think you missed the whole point of my response:
> Don't get your hopes up, it is very unlikely this will find its way into the software (its not even a bug since it currently works just fine)

and it was not:
> The API sucks, parsing HTML is much cooler

**Comment by chocolatkey** [comment]
@ChocolateOtaku I don't know how FMD works well enough to answer #4. I do not use it very often at all.

It seems the problem isn't that FMD isn't being developed, it's that the developer to user ratio is very low, and this is a free project. No official tech support, paid employees etc. Stuff will get implemented when people find time, and if they never do, then it never will. I am trying to make those few developers' lives easier by showing that comicake has an API as well, and that they don't need to scrape around.

Additionally, regarding the API: the response is also larger because it includes more rich data that the directory page does not. Compare sizes for properly compressed responses: https://whimsubs.xyz/api/comics.json = 6.9KB for 18 comics, https://whimsubs.xyz/r/directory/ = 5.0KB, for 12 comics
